python/src/graphql.rs

```diff
@@ -751,22 +731,143 @@ impl PyRaphtoryClient {
         }
     }
 
-    /// Set the server to load all the graphs from its path `path`.
+    /// Upload graph file from a path `file_path` on the client
+    ///
+    /// Arguments:
+    ///   * `name`: the name of the graph
+    ///   * `file_path`: the path of the graph on the client
+    ///   * `overwrite`: overwrite existing graph (defaults to False)
+    ///   * `namespace`: the namespace of the graph
+    ///
+    /// Returns:
+    ///    The `data` field from the graphQL response after executing the mutation.
+    #[pyo3(signature = (name, file_path, overwrite = false, namespace = None))]
+    fn upload_graph(
+        &self,
+        py: Python,
+        name: String,
+        file_path: String,
+        overwrite: bool,
+        namespace: Option<String>,
+    ) -> PyResult<HashMap<String, PyObject>> {
+        let rt = Runtime::new().unwrap();
+        rt.block_on(async {
+            let client = Client::new();
+
+            let mut file = File::open(Path::new(&file_path)).map_err(|err| adapt_err_value(&err))?;
+
+            let mut buffer = Vec::new();
+            file.read_to_end(&mut buffer).map_err(|err| adapt_err_value(&err))?;
+
+            let mut variables = format!(
+                r#""name": "{}", "overwrite": {}, "graph": null"#,
+                name, overwrite
+            );
+
+            if let Some(ns) = &namespace {
+                variables = format!(r#""namespace": "{}", {}"#, ns, variables);
+            }
+
+            let operations = format!(
```

Comment:
working with strings like this is error prone, have we tried some graphql typed libs

https://cynic-rs.dev/derives/query-fragments.html?highlight=muta#mutations
or
https://github.com/graphql-rust/graphql-client